app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -325,6 +319,25 @@ class ThreadFragment : Fragment() {
         threadAdapter.stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
     }
 
+    private fun onAttachmentClicked(attachable: Attachable) {
+        if (attachable is Attachment) {
+            trackAttachmentActionsEvent(ACTION_OPEN_NAME)
+            attachable.openAttachment(
+                context = requireContext(),
+                navigateToDownloadProgressDialog = { attachment, attachmentIntentType ->
+                    navigateToDownloadProgressDialog(
+                        attachment,
+                        attachmentIntentType,
+                        ThreadFragment::class.java.name,
+                    )
+                },
+                snackbarManager = snackbarManager,
+            )
+        } else if (attachable is SwissTransferFile) {
+            downloadSwissTransferFile(swissTransferFile = attachable)
+        }
```

Comment:
I personally like to have a when syntax for type checking because if you add another implementation of the interface, you won't have to transform the if in when

But I don't have a strong opinion on this one, you can asks the others also